### PR TITLE
feat: adaptor has its own adapter-mapping.xml

### LINKED ISSUE #337: <feature> 支持 koupleless-adapter 中自定义配置 adapter-mapping.yaml 规则
### Feature description

koupleless-base-build-plugin 会在基座打包时，根据 adapter-mapping.yaml 的规则匹配，找到基座依赖包对应的补丁包（koupleless-adapter-xxxx）。

其中，koupleless-base-build-plugin 在 [koupleless-runtime 仓库](https://github.com/koupleless/runtime/) 中；koupleless-adapter-xxxx 在 [koupleless-adapter 仓库](https://github.com/koupleless/adapter) 中。

目前，该配置（adapter-mapping.yaml）在 koupleless-base-build-plugin 中，这就导致了每个 adapter 新增/修改规则时，需要修改 koupleless-base-build-plugin 并发布新版本。

需求：
- 支持用户在 koupleless-adapter-xxxx 中配置 adapter-mapping.yaml 规则
- 支持 koupleless-base-build-plugin 读取到新的 adapter-mapping.yaml 规则

### PR DESCRIPTION
**Pull request by gaosaroma**
fix https://github.com/koupleless/koupleless/issues/337

## Summary by CodeRabbit

- **New Features**
	- Introduced new configuration files for various adapters, enhancing compatibility with specific versions of libraries such as Spring Boot, Log4j2, and RocketMQ.
	- Added new matcher mappings for Spring AOP and RocketMQ clients, facilitating precise version control.
	- Enhanced adapter mapping for Spring Boot Logback, allowing integration with specific versions of the Spring Boot framework.
	- Updated the `pom.xml` to include new modules for `koupleless-adapter-spring-boot-logback-2.7.14` and `koupleless-adapter-configs`.

- **Bug Fixes**
	- Updated comments in several classes for clarity regarding the koupleless adapter context.

- **Tests**
	- Added multiple test classes and methods to validate matcher functionality across different adapter versions and configurations.

- **Chores**
	- Updated `pom.xml` for versioning and module management, correcting typographical errors.

### CODE REVIEW

File: features/koupleless-adapter-test-utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        <version>${revision}</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>koupleless-adapter-test-utils</artifactId>
+    <version>0.0.1</version>
+
+    <properties>
+        <java.version>1.8</java.version>
+        <maven.deploy.skip>false</maven.deploy.skip>
```
Comment (lvjing2) [inline features/koupleless-adapter-test-utils/pom.xml:18]: 这个包应该只是本地构建需要用到，能否配置部署的时候跳过部署到中央仓库

File: koupleless-adapter-configs/.gitignore
Comment (lvjing2) [inline koupleless-adapter-configs/.gitignore:1]: 这个是否误提交了

File: features/koupleless-adapter-test-utils/pom.xml (hunk 1 above)
Comment (gaosaroma) [reply features/koupleless-adapter-test-utils/pom.xml:18]: > 这个包应该只是本地构建需要用到，能否配置部署的时候跳过部署到中央仓库

在 workflow 中跳过部署到中央仓库了

**Review by lvjing2 — APPROVED** [review]
LGTM

**Pull request merged**